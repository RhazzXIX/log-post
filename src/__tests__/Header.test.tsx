import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userevent from "@testing-library/user-event";
import Header from "@/component/Header";

const userInfo: IUser | undefined = {
  id: "125id",
  name: "test user",
  isAuthor: true,
};

const user = userevent.setup();

const loginFn = jest.fn();

const signUpFn = jest.fn();

describe("Header component", () => {
  it("Renders on screen.", () => {
    const { container } = render(<Header handleLogin= {loginFn} handleSignUp={signUpFn}/>);

    expect(container).toMatchSnapshot();
  });
  it("Shows appropriate buttons when the user is signed in.", () => {
    render(<Header user={userInfo} handleLogin= {loginFn} handleSignUp={signUpFn}/>);

    expect(screen.getByText("Profile")).toBeInTheDocument();
  });
  it("Activates the passed function when the appropriate button is clicked.", async () => {
    render(<Header handleLogin= {loginFn} handleSignUp={signUpFn} />);

    const loginBtn = screen.getByText("Log in");
    const signUpBtn = screen.getByText("Sign up");
    await user.click(loginBtn);

    expect(loginFn).toHaveBeenCalled();

    await user.click(signUpBtn);
    
    expect(signUpFn).toHaveBeenCalled();

    expect(loginFn).toHaveBeenCalledTimes(1);
    expect(signUpFn).toHaveBeenCalledTimes(1);
  });
});
