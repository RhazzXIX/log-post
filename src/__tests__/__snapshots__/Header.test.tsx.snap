// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Renders on screen. 1`] = `
<div>
  <header>
    <div
      class="widthLimiter"
    >
      <a
        href="/"
      >
        <h1>
          Log Pose
        </h1>
      </a>
      <button>
        Log in
      </button>
      <button>
        Sign up
      </button>
    </div>
  </header>
</div>
`;
